mono/mini/aot-runtime.c

```diff
@@ -2507,10 +2510,35 @@ load_aot_module (MonoAssemblyLoadContext *alc, MonoAssembly *assembly, gpointer
 
 	init_amodule_got (amodule, TRUE);
 
+#ifdef HOST_WASM
+	/*
+	 * Emscripten could reorder code so methods from different aot images are intermixed. Compute the continuous
+	 * regions of code and register each of them separately.
+	 */
+	sort_methods (amodule);
+	gpointer start = NULL;
+	for (i = 0; i < amodule->sorted_methods_len; ++i) {
+		//printf ("%s %d\n", amodule->assembly->aname.name, amodule->sorted_methods [i]);
+	}
+	for (i = 0; i < amodule->sorted_methods_len; ++i) {
+		if (start == NULL)
+			start = amodule->sorted_methods [i];
+		if (i > 0 && GPOINTER_TO_UINT (amodule->sorted_methods [i - 1]) + 1 != GPOINTER_TO_UINT (amodule->sorted_methods [i])) {
+			//printf ("%s %d %d\n", amodule->assembly->aname.name, start, amodule->sorted_methods [i - 1]);
+			mono_jit_info_add_aot_module (assembly->image, start, amodule->sorted_methods [i - 1]);
```

Comment:
this works because on wasm a function pointer is just an index in a table, right?  so the end `code_end` arg to `mono_jit_info_add_aot_module` is inclusive on wasm?

Reply:
Fixed.